### PR TITLE
Warn consistently for ``inplace`` in ``Series.rename``

### PR DESCRIPTION
**Pull request by phofl**
The previous warning seemed very inconsistent

### CODE REVIEW

**Review by jrbourbeau — COMMENTED** [review]
Thanks @phofl 

Can we safely remove the 

```python
                warnings.warn(
                    "'inplace' argument for dask series will be removed in future versions",
                    PendingDeprecationWarning,
                )
```

a few lines below now?

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -3923,6 +3923,14 @@ def rename(self, index=None, inplace=False, sorted_index=False):
 
         import dask.dataframe as dd
 
+        if inplace is not no_default:
+            warnings.warn(
+                "'inplace' argument for dask series will be removed in future versions",
+                PendingDeprecationWarning,
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask/dataframe/core.py]: updated

**Comment by phofl** [comment]
Yep good point, removed

**Review by hendrikmakait — APPROVED** [review]
Thanks, @phofl! This looks good to me, and all of @jrbourbeau's concerns seem to have been addressed.

**Pull request merged**